internal/services/point_service.go

```
@@ -264,23 +290,62 @@ func (p *PointWorker) taskProvider(job *models.Job, workChan chan models.CoinDBM
+		}
+		if len(vaults) == 0 {
+			break
+		}
+		for _, vault := range vaults {
+			for i := 0; i < len(p.cfg.GetCurrentSeason().Milestones); i++ {
+				// if total points is greater than or equal to milestone minimum
+				if vault.TotalPoints >= float64(p.cfg.GetCurrentSeason().Milestones[i].Minimum) {
+					// if this milestone is locked
+					if vault.NextMilestoneID <= i {
+						// unlock milestone: update vault total points and next milestone id
+						p.storage.UpdateVaultMilestone(vault.ID, i+1, float64(p.cfg.GetCurrentSeason().Milestones[i].Prize))
```

golangci-lint (1.64.8)

Error return value of `p.storage.UpdateVaultMilestone` is not checked

(errcheck)